tensorflow_datasets/structured/somerville_happiness.py

```diff
@@ -0,0 +1,88 @@
+"""Somerville Happiness Survey Data Set from UCI Machine Learning Repository"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import csv
+import tensorflow as tf
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """
+@misc{Waldemar:2020 ,
+author = "Waldemar W. Koczkodaj",
+year = "2015",
+title = "{UCI} Machine Learning Repository",
+url = "http://archive.ics.uci.edu/ml",
+institution = "University of California, Irvine, School of Information and Computer Sciences"
+}
+"""
+
+_DESCRIPTION = """It is a case of supervised learning with the use of Receiver Operating Characteristic (ROC) to 
+select the minimal set of attributes preserving or increasing predictability of the data.
+
+Attribute Information:
+
+D = decision attribute (D) with values 0 (unhappy) and 1 (happy) 
+X1 = the availability of information about the city services 
+X2 = the cost of housing 
+X3 = the overall quality of public schools 
+X4 = your trust in the local police 
+X5 = the maintenance of streets and sidewalks 
+X6 = the availability of social community events 
+
+Attributes X1 to X6 have values 1 to 5.
+"""
+_FEELING = ["happy", "unhappy"]
+
+_URL = 'https://archive.ics.uci.edu/ml/machine-learning-databases/00479/SomervilleHappinessSurvey2015.csv'
+
+class SomervilleHappiness(tfds.core.GeneratorBasedBuilder):
+  
+  
+  VERSION = tfds.core.Version('2.0.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        
+        features=tfds.features.FeaturesDict({
+           "feeling": tfds.features.ClassLabel(names=_FEELING),
+           "D": tfds.features.ClassLabel(num_classes=2),
+           "X1": tfds.features.ClassLabel(num_classes=5),
+           "X2": tfds.features.ClassLabel(num_classes=5),
+           "X3": tfds.features.ClassLabel(num_classes=5),
+           "X4": tfds.features.ClassLabel(num_classes=5),
+           "X5": tfds.features.ClassLabel(num_classes=5),
+           "X6": tfds.features.ClassLabel(num_classes=5),
+        }),
+        supervised_keys=("D", "feeling"),
```

Comment:
supervised_keys should be a 2-tuple structure (input, label).
You may want to define a FEATURES_DICT with all the features: D, X1, X2, X3, X4, X5, X6 and use that as "features" in your tfds.features.FeaturesDict here.
Then, supervised_keys=("features", "feeling") will be appropriate.